<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>lab4</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,1">
    <property name="spacing">
     <number>30</number>
    </property>
    <property name="leftMargin">
     <number>20</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <property name="rightMargin">
     <number>20</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <property name="spacing">
         <number>3</number>
        </property>
        <item row="1" column="1">
         <widget class="QPushButton" name="downPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>V</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QPushButton" name="rightPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>&gt;</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="horizontalPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>—</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QPushButton" name="leftPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>&lt;</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="upPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>^</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="verticalPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>|</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="upRightPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>/</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="upLeftPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>\</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGraphicsView" name="graphicsView"/>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="clearPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Очистить</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="recognizePushButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Распознать</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="spacing">
       <number>50</number>
      </property>
      <item>
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="verticalSpacing">
         <number>10</number>
        </property>
        <item row="1" column="1">
         <widget class="QSpinBox" name="numEpochSpinBox">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="maximum">
           <number>10000</number>
          </property>
          <property name="singleStep">
           <number>50</number>
          </property>
          <property name="value">
           <number>1000</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Норма обучения</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="learnRateSpinBox">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="decimals">
           <number>5</number>
          </property>
          <property name="minimum">
           <double>0.000000000000000</double>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>0.500000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Количество эпох</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QPushButton" name="learnPushButton">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Обучить</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>1</number>
        </property>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Результат распознавания:</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableView" name="recognizeTableView">
          <property name="font">
           <font>
            <family>Source Code Pro Light</family>
            <pointsize>14</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
